### DISCUSSION TITLE
Is FileUpload supposed to save itself?

### DISCUSSION THREAD

**Discussion by NicolasSacC** [filamentphp/filament#4209] [Help] [answered]
Might be a stupid question, but I have this code for my FileUpload:
```
            FileUpload::make('photo')
                ->image()
                ->disk('s3')
                ->directory('bags')
                ->visibility('private')
```

Since I put parameters for the disk and the directory, I expected it to save itself in some way. However, when I dd($this->photo), I end up with the Livewire\TemporaryUploadedFile. 

How can I save the file at the proper place? Is this something that Filament does by default?

**Comment by ManojKiranA** [comment, marked as answer]
File uploads are handled by filament itself. It will save in the specified disk.

**Reply by NicolasSacC** [reply]
How? When? 

I tried to save it to my model and it does nothing. Do you have a code example on how to save the file?

**Reply by danharrin** [reply]
`$this->form->getState()` will validate the form including uploaded files, upload files, and return the form data in an array. Never read from a Livewire property, always from that array.
